In [5]:
import torch

In [2]:
a = [1,2,3]

In [3]:
## 
b = torch.ones(3)

In [4]:
b

tensor([1., 1., 1.])

In [5]:
c = torch.ones(3,4)

In [6]:
c 

tensor([[1., 1., 1., 1.],
        [1., 1., 1., 1.],
        [1., 1., 1., 1.]])

In [7]:
a[1]

2

In [8]:
a[0]

1

In [10]:
float(a[1])

2.0

In [15]:
a[2]=2.0
a

[1, 2, 2.0]

In [16]:
c[2] = 100
c

tensor([[  1.,   1.,   1.,   1.],
        [  1.,   1.,   1.,   1.],
        [100., 100., 100., 100.]])

In [17]:
a=1
t1=torch.tensor(a)
t1

tensor(1)

In [18]:
b=[1,2,3]

In [20]:
t2 = torch.tensor(b)
t2

tensor([1, 2, 3])

In [24]:
import numpy as np

In [26]:
arr = np.array([[1,2,3],[4,5,6]])

In [30]:
t3 = torch.from_numpy(arr)
t3

tensor([[1, 2, 3],
        [4, 5, 6]])

In [28]:
t4 = torch.from_numpy(arr)
t4

tensor([[1, 2, 3],
        [4, 5, 6]])

In [31]:
arr[0][0]=100
arr

array([[100,   2,   3],
       [  4,   5,   6]])

In [32]:
##传入张量的形状
t=torch.zeros(3,3)
t

tensor([[0., 0., 0.],
        [0., 0., 0.],
        [0., 0., 0.]])

In [35]:
b = torch.zeros_like(t3)
b

tensor([[0, 0, 0],
        [0, 0, 0]])

In [6]:
e = torch.full([3,3], 4)
e 

tensor([[4, 4, 4],
        [4, 4, 4],
        [4, 4, 4]])

In [7]:
import torch
 
print(torch.__version__) #注意是双下划线

1.11.0


In [8]:
print(torch.__version__)

1.11.0


In [9]:
import sys
print(sys.version)

3.7.16 (default, Jan 17 2023, 09:28:58) 
[Clang 14.0.6 ]


```
530  conda create -n pytorch
  531  source active pytorch
  532  cd envs/
  533  source activate pytorch
  534  python
  535  python3
  536  python -m ipykernel install --name pytorch
  537  pip install ipykernel
  538  pip3 install ipykernel
  539  python -m ipykernel install --name pytorch
  540  python3
  541  python -m ipykernel install --user --name base --display-name "Python [conda env:base]"
  542  conda install python=3.7
  543  conda install python=3.7.4
  544  conda update python
  545  conda config
  546  vi ~/.condarc
  547  conda install python=3.7.4
  548  conda install python=3.7
  549  vi ~/.condarc
  550  cp ~/.condarc ~/.condarc-bak
  551  > ~/.condarc
  552  vim ~/.condarc
  553  conda install python=3.7
  554  python3
  555  clear
  556  history
  557  python -m ipykernel install --user --name pytorch --display-name "Python [conda env:pytorch]"
  558  pip3 install ipykernel
  559  python -m ipykernel install --user --name pytorch --display-name "Python [conda env:pytorch]"
  560  history
  561  conda install pytorch==1.11.0 torchvision==0.12.0 torchaudio==0.11.0 -c pytorch
  562  history
  563  clear
  564  history